src/build_system.ml

```diff
@@ -398,7 +399,15 @@ let compile_rule t ~all_targets_by_dir ?(allow_override=false) pre_rule =
     in
     let targets_min_ts = min_timestamp t targets_as_list  in
     let deps_max_ts    = max_timestamp t all_deps_as_list in
-    if rule_changed ||
+    let is_runtest ps  =
+      let is_runtest p =
+        match Path.extract_build_context p with
+        | Some (".aliases", _) -> true
```

Comment:
This matches any aliases, you also need to check that the basename is of the form `runtest-...`. While we are at it, let's add a function `Utils.alias_of_file : Path.t -> (Path.t * string) option`. This is used in a few places